src/utils/nag-suppression-helper.ts

```diff
@@ -72,6 +79,37 @@ export class NagSuppressionHelper {
     }
   }
 
+  static doesApply(
+    suppression: NagPackSuppression,
+    ruleId: string,
+    findingId: string
+  ): boolean {
+    if (ruleId !== suppression.id) {
+      return false;
+    }
+
+    if (!suppression.appliesTo) {
+      // the rule is not granular so it always applies
+      return true;
+    }
+
+    if (
+      findingId &&
+      suppression.appliesTo.some((appliesTo) => {
+        if (typeof appliesTo === 'string') {
```

Comment:
I'm not sure about this logic, I don't think we should try to infer whether string or regex especially since the values can change as the application changes. You could have a scenario where an intended string match unintentionally turns into a regex match if the finding is remediated. 

Let's say as a user you temporarily suppress an error on `s3:*`. Later on `s3:*` gets swapped for `s3:Get*`. You may not actually want `s3:Get*` to be allowed, however the `match` method will flag that as a match and suppress the warning/error.

Reply:
Since `appliesTo` will be used for other things (ports, log configurations, unknowns) in the future there could be some other edge cases to consider which would be avoided by being explicitly requiring the user to say that they want to use regex.

Reply:
> You could have a scenario where an intended string match unintentionally turns into a regex match if the finding is remediated.

I might be misunderstanding what you're saying but we're not testing if a string is a regex here.

This `typeof` test simply determines whether the `appliesTo` in in the "old-school" string form like `Action::s3:*` in which case the `typeof` is `string` or if the `appliesTo` is the new form like `{ regex: "...some regex..." )` in which case the `typeof` is `object`.

So with what I said above I don't see how that can happen since the user has to define exactly what it is, a regex or an exact string match.

Reply:
Scratch my previous comment .. I somehow interpreted this as the `appliesTo` string being conditionally treated as a Regex.